Hakeem/StepperMotor

Hakeem098 opened the pull request.

LiamDroog commented on devices/StepperMotor/Threading_StepperMotor.py:
```diff
@@ -0,0 +1,522 @@
+from time import sleep
+import RPi.GPIO as GPIO
+
+from PyQt5.QtGui import *
```
Should import only functions that are used within the file, import * isn't a great practice.

LiamDroog commented on devices/StepperMotor/Threading_StepperMotor.py:
```diff
@@ -0,0 +1,522 @@
+from devices.StepperMotor.Hakeem_StepperMotors import (Ui_Form) 
+
+# Define GPIO pins and constants
+global DIR, STEP, CW, CCW, log
```
Also we shouldn't use global variables, three letter variables can quickly get overwritten when this goes into a larger diagnostic system and that will be a nightmare to debug

LiamDroog reviewed (commented):
Everything in Threading_StepperMotor.py should be documented (classes, functions) to at least a version of: "This is what the function does, these are the expected inputs for the function, this is what the function outputs, these are the error states that might arise"

It'd also be good to include a drawn / computer drawn circuit diagram for posterity down the road if you have the time.

Hakeem098 commented:
Thanks for the comments Liam. The issues have been resolved. An SOP will be
created by me for future reference.

On Thu, Feb 27, 2025 at 10:18 AM Liam Droog ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
>
> Everything in Threading_StepperMotor.py should be documented (classes,
> functions) to at least a version of: "This is what the function does, these
> are the expected inputs for the function, this is what the function
> outputs, these are the error states that might arise"
>
> It'd also be good to include a drawn / computer drawn circuit diagram for
> posterity down the road if you have the time.
> ------------------------------
>
> In devices/StepperMotor/Threading_StepperMotor.py
> <https://github.com/nfbeier/15tw-smartsystem/pull/22#discussion_r1974020196>
> :
>
> > @@ -0,0 +1,522 @@
> +from PyQt5 import QtWidgets, QtCore, QtGui
> +import sys, os
> +from time import sleep
> +import RPi.GPIO as GPIO
> +
> +from PyQt5.QtGui import *
>
> Should import only functions that are used within the file, import * isn't
> a great practice.
> ------------------------------
>
> In devices/StepperMotor/Threading_StepperMotor.py
> <https://github.com/nfbeier/15tw-smartsystem/pull/22#discussion_r1974021536>
> :
>
> > +print(cwd)
> +# Check if '15tw-smartsystem' is in the components
> +if "15tw-smartsystem" not in cwd.split(os.path.sep):
> +    raise ValueError("The directory does not contain '15tw-smartsystem' folder.")
> +
> +# Rebuild the directory string up to and including '15tw-smartsystem', prevent import errors
> +cwd = os.path.sep.join(
> +    cwd.split(os.path.sep)[: cwd.split(os.path.sep).index("15tw-smartsystem") + 1]
> +)
> +sys.path.insert(0, cwd)
> +
> +
> +from devices.StepperMotor.Hakeem_StepperMotors import (Ui_Form)
> +
> +# Define GPIO pins and constants
> +global DIR, STEP, CW, CCW, log
>
> Also we shouldn't use global variables, three letter variables can quickly
> get overwritten when this goes into a larger diagnostic system and that
> will be a nightmare to debug
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nfbeier/15tw-smartsystem/pull/22#pullrequestreview-2648516702>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5TQZCP7MRSWTSMYMTODXIL2R5CIDAVCNFSM6AAAAABYAK7WA2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMNBYGUYTMNZQGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
